### DISCUSSION TITLE
How to bind a collection?

### DISCUSSION THREAD

**Discussion by LouisIngenthron** [AvaloniaUI/Avalonia#17711] [Q&A] [answered]
The doc page for binding a collection (https://docs.avaloniaui.net/docs/guides/data-binding/how-to-bind-to-a-collection) references a "SetProperty" method that apparently doesn't exist anymore (at least my compiler can't find it).

So how are we supposed to bind collections now?  Without that, I can't seem to get anything to show on the view.

Edit: Apparently there's no such thing as an "ObservableObject" either?  What exactly are the docs supposed to be documenting?

**Comment by stevemonaco** [comment, marked as answer]
You likely skimmed over a [previous section](https://docs.avaloniaui.net/docs/guides/data-binding/inotifypropertychanged#using-mvvm-toolkit-to-simplify-inotifypropertychanged).

It comes from Mvvm Toolkit which is the recommended MVVM framework, especially for beginners.

Keep in mind that nothing consumed in the ViewModel should come from Avalonia (barring intentional breaking of MVVM rules). The ViewModel shouldn't refer to Avalonia types as VMs are in theory directly reusable in other GUI/TUI frameworks. The docs occasionally break this rule for the sake of simplicity though.

**Reply by LouisIngenthron** [reply]
Ah.  [This page](https://docs.avaloniaui.net/docs/concepts/the-mvvm-pattern) had convinced me that MVVM was for larger apps and would be overkill for someone just beginning.

It would be nice if the docs could somehow mention when their topic only applies to one paradigm and not the other (and why).

**Reply by stevemonaco** [reply]
If you aren't using MVVM, then the amount of binding you need to do is drastically reduced. There's a much larger amount of code in code-behind because you're writing code to directly update control state. eg. Assigning `someListBox.ItemsSource = YourCollection;` in an event handler or in the constructor instead of writing VMs and a binding.

Personally, I always use MVVM even for apps that are only a few hundred LoC. I may flex the rules more on smaller apps. The only time I consider not using MVVM is when writing samples/repros to reduce the amount of code/classes the other party needs to read, when reasonable.
